Foundation/URLCache.swift

```diff
@@ -154,6 +195,9 @@ open class CachedURLResponse : NSObject, NSCopying {
 
 open class URLCache : NSObject {
     
+    private static let sharedLock = NSLock()
+    private static var _shared = URLCache(memoryCapacity: 4 * 1024 * 1024, diskCapacity: 20 * 1024 * 1024, diskPath: nil)
```

Comment:
@drodriguez IIRC the language guarantees that, for static and global variables, their initializers aren't run until first access anyway.

Reply:
Yes, that's true, but that initialization seems to happen even if the static variable is a lvalue, like in line 226.

You can type the following in an Xcode playground and check:

```swift
var globalCounter = 0

class TestValue {
    init() {
        print("TestValue.init")
        globalCounter += 1
    }
}

class TestClass {
    private static var _static: TestValue = TestValue()
    
    class var accessor: TestValue {
        get {
            print("TestClass.accessor.get: before return")
            return _static
        }
        set {
            print("TestClass.accessor.set: before assign")
            _static = newValue
            print("TestClass.accessor.set: after assign")
        }
    }
}

let t = TestValue()
TestClass.accessor = t
print(globalCounter)

// Output:
// TestValue.init
// TestClass.accessor.set: before assign
// TestValue.init
// TestClass.accessor.set: after assign
// 2
```